zerver/lib/retention.py

```diff
@@ -276,7 +276,8 @@ def archive_stream_messages(realm: Realm, chunk_size: int=MESSAGE_BATCH_SIZE) ->
     logger.info("Archiving stream messages for realm " + realm.string_id)
     # We don't archive, if the stream has message_retention_days set to -1,
     # or if neither the stream nor the realm have a retention policy.
-    streams = Stream.objects.filter(realm_id=realm.id).exclude(message_retention_days=-1)
+    streams = Stream.objects.filter(realm_id=realm.id).exclude(message_retention_days=-1
+                                                               ).select_related("recipient")
```

Comment:
This line-wrapping is awkward; it's better to move the `.select_related()` earlier.

Reply:
This also made me realize we have a lot of places that currently do `Stream.objects.select_related("realm")...` when fetching Stream objects that should be fixed.  I'll just add an extra commit to fix those and merge that with this PR.